connector/sumconnector/config.go

```diff
@@ -21,6 +32,91 @@ type MetricInfo struct {
 }
 
 type AttributeConfig struct {
-	Key          string `mapstructure:"key"`
+	AttrKey      string `mapstructure:"key"`
 	DefaultValue any    `mapstructure:"default_value"`
 }
+
+func (c *Config) Validate() (errors error) {
+	for name, info := range c.Spans {
+		if name == "" {
+			errors = multierr.Append(errors, fmt.Errorf("spans: metric name missing"))
+		}
+		if _, err := filterottl.NewBoolExprForSpan(info.Conditions, filterottl.StandardSpanFuncs(), ottl.PropagateError, component.TelemetrySettings{Logger: zap.NewNop()}); err != nil {
+			errors = multierr.Append(errors, fmt.Errorf("spans condition: metric %q: %w", name, err))
+		}
+		if err := info.validateAttributes(); err != nil {
+			errors = multierr.Append(errors, fmt.Errorf("spans attributes: metric %q: %w", name, err))
+		}
+		if info.SourceAttribute == "" {
```

Comment:
Minor nit: It would be slightly easier to follow for me if the simple checks are first for the required options, then do the optional checks. Specifically, check `SourceAttribute` before the ottl and attribute validation checks.

_Same comment for other contexts below_